core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -266,12 +280,16 @@ private CloseableIterable<ManifestEntry<F>> open(Schema projection) {
   }
 
   CloseableIterable<ManifestEntry<F>> liveEntries() {
-    return CloseableIterable.filter(
-        content == FileType.DATA_FILES
-            ? scanMetrics.skippedDataFiles()
-            : scanMetrics.skippedDeleteFiles(),
-        entries(),
-        entry -> entry != null && entry.status() != ManifestEntry.Status.DELETED);
+    return entries(true /* only live entries */);
+  }
+
+  private CloseableIterable<ManifestEntry<F>> filterLiveEntries(
```

Comment:
Minor: these last two methods could be static.

Reply:
I did it this way so that I don't have to add `F extends ContentFile<F>` to these static methods and reuse the bound already defined in the class. I could use `?` in the second one but not here.